homeassistant/components/volumio/browse_media.py

```diff
@@ -100,11 +101,13 @@ def _list_payload(media_library, item, children=None):
     )
 
 
-def _raw_item_payload(media_library, item, parent_item=None, title=None, info=None):
+def _raw_item_payload(entity, item, parent_item=None, title=None, info=None):
     if "type" in item:
         thumbnail = item.get("albumart")
         if thumbnail:
-            thumbnail = media_library.canonic_url(thumbnail)
+            # we double encode because yarl is over-zealous with decoding
+            encoded_url = quote_plus(quote_plus(thumbnail))
+            thumbnail = entity.get_browse_image_url(MEDIA_TYPE_MUSIC, encoded_url)
```

Comment:
passing image urls isn't desirable for this api as it could create an open proxy. you'll need to store the url and pass a reference id like the media_id instead. plex has a good example of this. https://github.com/home-assistant/core/blob/dev/homeassistant/components/plex/media_browser.py#L74

Reply:
its also best to check for internal request before proxying the image when not needed. also example of that in the same plex example above

Reply:
Thanks @ctalkington. I see that Plex is creating a cache, but it's not invalidated anywhere, which makes it a memory leak. Is that behavior ok?

As for internal request - not exactly sure what you mean. If the HA request is over HTTPS, it will fail even with internal requests.

Reply:
thats a fair point, not sure we have a better caching strategy in core at this time. Maybe some LRU type setup would be ideal. How many media entries do common users have?

Reply:
Don't know about common users, but I have several hundreds of albums and several hundreds of artists, and I don't believe it's considered a large collection. On the other hand, even if we assume 250 bytes per entry (which I believe is an over estimation), and tens of thousands of entries, it's still not a huge deal (50,000 entries, which I believe is an enormous collection, will only be about 12MB).
I guess I just need to get over the psychological barrier of contributing a deliberate memory leak :)

Reply:
@ctalkington I added the ~~memory leak~~ thumbnail cache :)